### PR TITLE
Adjust reward model's score module and pooler module order for reducing computation

### PR DESCRIPTION
**Pull request by aqweteddy**
## Motivation

* @merrymercy mentioned in #1954 
* remove redundant code in gemma2 reward model.

## Modifications

* Adjust the order of the `pooler` and `score` modules in `class LlamaForSequenceClassification` and `class Gemma2ForSequenceClassification` to reduce computation.
* Remove redundant model-loading code in gemma2 reward model.

## Checklist

- [V] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [V] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [V] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

File: python/sglang/srt/models/gemma2_reward.py [hunk 1]
```diff
@@ -58,43 +58,11 @@ def forward(
...
 
         hidden_states = self.model(input_ids, positions, forward_batch, input_embeds)
-        scores = self.score(hidden_states)
+        last_token_hidden = self.pooler(hidden_states, forward_batch).embeddings
+        scores = self.score(last_token_hidden)
 
-        return self.pooler(scores, forward_batch)
+        return EmbeddingPoolerOutput(scores)
 
-    def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
```
Comment (merrymercy) [inline python/sglang/srt/models/gemma2_reward.py:65]: can you also simplify the weight loader of `LlamaForSequenceClassification`?

**Comment by merrymercy** [comment]
Can you fix the lint error?

File: python/sglang/srt/models/gemma2_reward.py [hunk 2]
```diff
@@ -58,43 +58,11 @@ def forward(
...
-                # lm_head is not used in vllm as it is tied with embed_token.
-                # To prevent errors, skip loading lm_head.weight.
-                if "lm_head.weight" in name:
-                    continue
-                # Skip loading extra bias for GPTQ models.
-                if name.endswith(".bias") and name not in params_dict:
-                    continue
-                param = params_dict[name]
-                weight_loader = getattr(param, "weight_loader", default_weight_loader)
-                weight_loader(param, loaded_weight)
```
Comment (merrymercy) [inline python/sglang/srt/models/gemma2_reward.py]:
```suggestion
```

File: python/sglang/srt/models/gemma2_reward.py (hunk 1 above)
Comment (aqweteddy) [reply python/sglang/srt/models/gemma2_reward.py:65]: Done & verified.
image

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 default_language_version:
-    python: python3.9
+    python: python3.10
```
Comment (merrymercy) [inline .pre-commit-config.yaml]:
```suggestion
    python: python3.9
```

**Review by merrymercy — APPROVED** [review]

**Comment by merrymercy** [comment]
@aqweteddy Thanks for the contribution. It is merged.

**Pull request merged**